src/main/java/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,29 @@
+package command;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import task.*;
+import duke.*;
+
+public class AddDeadlineCommand extends AddCommand {
+
+    private String desc;
+    private LocalDate by;
+
+    public AddDeadlineCommand(String desc, String by) {
```

Comment:
LocalDate.parse can throw a DateTimeException, should either catch it here or make your function explicitly throw the exception.

```suggestion
    public AddDeadlineCommand(String desc, String by) throws DateTimeException {
```